### PR TITLE
feat: Add lib entry point comments

### PR DESCRIPTION
**Pull request by chrismaree**
This PR adds entry point comments to lib.rs. there is no "natspec" for solana so I came up with a structure that I think covers enough detail on accounts and params without being too wordy. inspired from SpokePool.sol

### CODE REVIEW

File: programs/svm-spoke/src/lib.rs [hunk 1]
```diff
@@ -2,10 +2,18 @@ use anchor_lang::prelude::*;
 
 declare_id!("Fdedr2RqfufUiE1sbVEfpSQ3NADJqxrvu1zojWpQJj4q");
 
-// External programs from idls directory (requires `anchor run generateExternalTypes`).
+// External programs from idls directory (requires anchor run generateExternalTypes).
 declare_program!(message_transmitter);
 declare_program!(token_messenger_minter);
 
+/// # Across SVM Spoke Prgram
```
Comment (Reinis-FRP) [inline programs/svm-spoke/src/lib.rs:9]: nit: shall it better go below just above the `#[program]` macro? not sure

File: programs/svm-spoke/src/lib.rs [hunk 2]
```diff
@@ -2,10 +2,18 @@ use anchor_lang::prelude::*;
...
+///
+/// Spoke pool implementation for Across Protocol enabling connection to the Solana Ecosystem. Program is functionally
+/// the re-implementation of SpokePool.sol for Solana, with some extensions to be Solana compatible. The implementation
+/// Leverages Circle's CCTP for message and token bridging back and forth from Ethereum mainnet. As the EVM spoke pool,
+/// this spoke pool is instructed by the EVM hubpool for pool rebalancing and relayer repayment.
+///
+/// For any issues, please reach out to bugs@across.to.
```
[the remark this answers is not in the document]
Comment (chrismaree) [reply programs/svm-spoke/src/lib.rs:16]: yup. https://solana.com/ru/developers/guides/advanced/verified-builds#how-to-implement-securitytxt

File: programs/svm-spoke/src/lib.rs (hunk 1 above)
Comment (chrismaree) [reply programs/svm-spoke/src/lib.rs:9]: from the research I did it should be right below the `daclare_id` but before `#[program]` just like the `* @title` solidity block goes before the contract entry point.

File: programs/svm-spoke/src/instructions/admin.rs [hunk 3]
```diff
@@ -62,7 +62,7 @@ pub fn initialize(
 #[event_cpi]
 #[derive(Accounts)]
 pub struct PauseDeposits<'info> {
-    #[account(constraint = is_local_or_remote_owner(&signer, &state) @ SvmError::NotOwner)]
+    #[account(constraint = is_local_or_remote_owner(&signer, &state) @ SvmError::NotOwner)] // Enforces onlyOwner.
```
Comment (md0x) [inline programs/svm-spoke/src/instructions/admin.rs:65]: Nit: maybe overly verbose here

Comment (chrismaree) [reply programs/svm-spoke/src/instructions/admin.rs:65]: ye, fair. will remove.

**Review by md0x — COMMENTED** [review]
We should consider adding comments for the remaining fields here:
https://github.com/across-protocol/contracts/blob/45f4e9c47b4017db2cb4b94b8293ab54b32f9902/programs/svm-spoke/src/state/state.rs#L6-L8

**Comment by chrismaree** [comment]
> We should consider adding comments for the remaining fields here:
> 
> https://github.com/across-protocol/contracts/blob/45f4e9c47b4017db2cb4b94b8293ab54b32f9902/programs/svm-spoke/src/state/state.rs#L6-L8

good shout. I'll do this rn.

**Comment by chrismaree** [comment]
> > We should consider adding comments for the remaining fields here:
> > https://github.com/across-protocol/contracts/blob/45f4e9c47b4017db2cb4b94b8293ab54b32f9902/programs/svm-spoke/src/state/state.rs#L6-L8
> 
> good shout. I'll do this rn.

done on all state.

File: programs/svm-spoke/src/instructions/handle_receive_message.rs [hunk 4]
```diff
@@ -7,15 +7,14 @@ use crate::{
...
     utils::{self, EncodeInstructionData},
-    State,
 };
 
 #[derive(Accounts)]
 #[instruction(params: HandleReceiveMessageParams)]
 pub struct HandleReceiveMessage<'info> {
-    // authority_pda is a Signer to ensure that this instruction
-    // can only be called by Message Transmitter
+    // authority_pda is a Signer to ensure that this instruction. Can only be called by Message Transmitter
```
Comment (md0x) [inline programs/svm-spoke/src/instructions/handle_receive_message.rs:17]:
```suggestion
    // authority_pda is a Signer to ensure that this instruction can only be called by the Message Transmitter.
```

File: programs/svm-spoke/src/lib.rs (hunk 1 above)
Comment (md0x) [reply programs/svm-spoke/src/lib.rs:9]:
```suggestion
/// # Across SVM Spoke Program
```

File: programs/svm-spoke/src/lib.rs [hunk 5]
```diff
@@ -2,10 +2,18 @@ use anchor_lang::prelude::*;
...
-// External programs from idls directory (requires `anchor run generateExternalTypes`).
+// External programs from idls directory (requires anchor run generateExternalTypes).
 declare_program!(message_transmitter);
 declare_program!(token_messenger_minter);
 
+/// # Across SVM Spoke Prgram
+///
+/// Spoke pool implementation for Across Protocol enabling connection to the Solana Ecosystem. Program is functionally
+/// the re-implementation of SpokePool.sol for Solana, with some extensions to be Solana compatible. The implementation
+/// Leverages Circle's CCTP for message and token bridging back and forth from Ethereum mainnet. As the EVM spoke pool,
```
Comment (md0x) [inline programs/svm-spoke/src/lib.rs:13]:
```suggestion
/// leverages Circle's CCTP for message and token bridging back and forth from Ethereum mainnet. As the EVM spoke pool,
```

File: programs/svm-spoke/src/lib.rs [hunk 6]
```diff
@@ -104,11 +137,113 @@ pub mod svm_spoke {
...
+    ///
+    /// The fee paid to relayers and the system should be is captured in the spread between input and output amount
+    /// when adjusted to be denominated in the input token. A relayer on the destination chain will send output_amount
+    /// of output_token to the recipient and receive input_tokenon a repayment chain of their choice. Therefore,
+    /// the fee should account for destination fee transaction costs, the relayer's opportunity cost of capital while
+    /// they wait to be refunded following an optimistic challenge window in the HubPool, and the system fee that
+    /// they'll be charged. On the destination chain, the hash of the deposit data will be used to uniquely identify
+    /// this deposit, so modifying any params in it will result in a different hash and a different deposit. The hash
+    /// will comprise all parameters to this function along with this chain's chainId(). Relayers are only refunded for
+    /// filling deposits with deposit hashes that map exactly to the one emitted by this contract.
```
Comment (md0x) [inline programs/svm-spoke/src/lib.rs:209]: Improve readability and fix some typos
```suggestion
    /// The fee paid to relayers and the system is captured in the spread between the input and output amounts,
    /// denominated in the input token. A relayer on the destination chain will send `output_amount` of `output_token`
    /// to the recipient and receive `input_token` on a repayment chain of their choice.
    ///
    /// The fee accounts for:
    /// - Destination transaction costs,
    /// - The relayer's opportunity cost of capital while waiting for a refund during the optimistic challenge window in the HubPool,
    /// - The system fee charged to the relayer.
    ///
    /// On the destination chain, a unique hash of the deposit data is used to identify this deposit. Modifying any
    /// parameters will result in a different hash, creating a separate deposit. The hash is computed using all parameters
    /// of this function along with the chain's `chainId()`. Relayers are refunded only for deposits with hashes that
    /// exactly match those emitted by this contract.
```

File: programs/svm-spoke/src/lib.rs [hunk 7]
```diff
@@ -104,11 +137,113 @@ pub mod svm_spoke {
...
+    ///
+    /// ### Accounts:
+    /// - signer (Signer): The account that authorizes the deposit.
+    /// - state (Writable): Spoke state PDA. Seed: ["state", seed] where seed is 0 on mainnet.
+    /// - route (Account): The route PDA for the particular bridged route in question. Validates a route is enabled.
+    ///   Seed: ["route", input_token, state.seed, destination_chain_id].
+    /// - depositor_token_account (Writable): The depositor's ATA for the input token.
+    /// - vault (Writable): Programs ATA for the associated input token. This is where the depositors assets are sent.
```
Comment (md0x) [inline programs/svm-spoke/src/lib.rs:217]:
```suggestion
    /// - vault (Writable): Programs ATA for the associated input token. This is where the depositor's assets are sent.
```

**Review by md0x — APPROVED** [review]
